.github/workflows/checkov-scan.yml

```diff
@@ -0,0 +1,74 @@
+name: Checkov Security Scan
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+permissions: read-all
+jobs:
+  security_scan:
+    runs-on: ubuntu-latest
+
+    steps:
+      ### Step 1: Checkout Repo
+      - name: _Checkout PR Code
+        uses: actions/checkout@v3
+
+      ### Step 2: Run Checkov Scan 
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: _Install Checkov
+        run: pip install checkov
+      - name: _Install jq
+        run: sudo apt-get install jq
+
+      - name: _Run Checkov Scan (JSON)
+        run: |
+          echo "PWD: $(pwd)"
+          SKIP_PATHS=$(find . -type d -iname "*test*" | sed 's/^/--skip-path /' | tr '\n' ' ')
+          echo "Using SKIP_PATHS: $SKIP_PATHS"
+          checkov -d . $SKIP_PATHS \
+          --skip-path ./.github/checkov-scan/checkov_raw_report.json \
+          --output json > ./.github/checkov-scan/checkov_raw_report.json || true
+
+      ### Step 3: Build and Run Go code to enrich severity
+      - name: _Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.24'
```

Comment:
This should not be required if we remove the checkov-tool-utility code from this repo.